src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -61,11 +64,15 @@ public static int Main(string[] args)
                 if (arg == "--debug-agent")
                 {
                     if (!Debugger.IsAttached)
-                        Debugger.Launch();
+                        TryLaunchDebugger();
                 }
```

Comment:
Rather than depending on the order of args in the call in TestAgency, what if you merely set flags in this switch statement and pushed all the stuff that may generate an error down after the switch?